operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaCrdOperatorIT.java

```diff
@@ -0,0 +1,174 @@
+                    .endZookeeper()
+                .endSpec()
+                .withNewStatus()
+                .endStatus()
+                .build();
+    }
+
+    @Test
+    public void testUpdateStatus(TestContext context)    {
+        log.info("Getting Kubernetes version");
+        Async versionAsync = context.async();
+        PlatformFeaturesAvailability.create(vertx, client).setHandler(pfaRes -> {
+            if (pfaRes.succeeded())    {
+                PlatformFeaturesAvailability pfa = pfaRes.result();
+                context.put("pfa", pfa);
+                versionAsync.complete();
+            } else {
+                context.fail(pfaRes.cause());
+            }
+        });
+        versionAsync.awaitSuccess();
+
+        if (((PlatformFeaturesAvailability) context.get("pfa")).getKubernetesVersion().compareTo(KubernetesVersion.V1_11) < 0) {
+            log.info("Kubernetes {} is too old", ((PlatformFeaturesAvailability) context.get("pfa")).getKubernetesVersion());
+            return;
+        }
+
+        log.info("Creating operator");
+        CrdOperator<KubernetesClient, Kafka, KafkaList, DoneableKafka> kafkaOperator = new CrdOperator(vertx, client, Kafka.class, KafkaList.class, DoneableKafka.class);
+
+        log.info("Creating resource");
+        Async createAsync = context.async();
+        kafkaOperator.reconcile(namespace, RESOURCE_NAME, getResource()).setHandler(res -> {
+            if (res.succeeded())    {
+                createAsync.complete();
+            } else {
+                context.fail(res.cause());
+            }
+        });
+        createAsync.awaitSuccess();
+
+        Kafka withStatus = new KafkaBuilder(kafkaOperator.get(namespace, RESOURCE_NAME))
+                .withNewStatus()
+                .withConditions(new ConditionBuilder()
+                        .withType("Ready")
+                        .withStatus("True")
+                        .build())
+                .endStatus()
+                .build();
+
+        log.info("Updating resource status");
+        Async updateStatusAsync = context.async();
+        kafkaOperator.updateStatusAsync(withStatus).setHandler(res -> {
+            if (res.succeeded())    {
+                kafkaOperator.getAsync(namespace, RESOURCE_NAME).setHandler(res2 -> {
+                    if (res2.succeeded())    {
+                        Kafka updated = res2.result();
+
+                        context.assertEquals("Ready", updated.getStatus().getConditions().get(0).getType());
+                        context.assertEquals("True", updated.getStatus().getConditions().get(0).getStatus());
+
+                        updateStatusAsync.complete();
+                    } else {
+                        context.fail(res.cause());
+                    }
+                });
+            } else {
+                context.fail(res.cause());
+            }
+        });
+        updateStatusAsync.awaitSuccess();
+    }
+}
```

Comment:
I think we want a test here for the concurrent modification case, and maybe also the concurrent deletion case.